### PR TITLE
bugfix: when only Lua preread handler is present and content handler is

### PR DESCRIPTION
**Pull request by dndx**
non-Lua, cleanup hooks will be incorrectly skipped. #4d0361f

Changes were generated from https://github.com/openresty/meta-lua-nginx-module/pull/9.

To reproduce, simply run the added test without this PR using no-pool NGINX.

`stream` subsystem does not support cleanup handler natively, so we added emulation for such functionality inside `stream_lua`. However, those cleanup handlers will only be executed if "request" is terminated using `ngx_stream_lua_finalize_real_request()`. Since this function is `stream_lua` specific, content phase handlers like the built-in `return` directive will **not** execute those cleanup hooks.

Later on when the `__gc` metamethod on the cosocket udata is executed in another "request", it will attempt to dereference the old "request" which would have been freed already, thus causing the segfault.

This PR attaches the cleanup handlers to the connection's pool cleanup list to ensure they will always be executed regardless which content handler the server uses. Later on when Lua decides to GC the udata the `__gc` metamethod will notice that `u->cleanup` is already `NULL` and will not attempt to run the finalizer (and dereference the "request").

### CODE REVIEW

File: t/058-tcp-socket.t [hunk 1]
```diff
@@ -3087,3 +3087,39 @@ failed to setkeepalive: closed
 failed to connect: bad port number: 65536
 --- no_error_log
 [error]
+
+
```
Comment (agentzh) [inline t/058-tcp-socket.t]: Please use `reindex` to re-format your `.t` files.

Comment (dndx) [reply t/058-tcp-socket.t]: Done.

File: t/075-logby.t [hunk 2]
```diff
@@ -204,7 +204,7 @@ API disabled in the context of log_by_lua*
 
 
 
-=== TEST 14: no ngx.flush
+=== TEST 12: no ngx.flush
```
Comment (agentzh) [inline t/075-logby.t]: These changes are unrelated and should go to separate PRs to make this PR atomic.
